src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -483,9 +492,14 @@ bool D4NFilterObject::check_head_exists_in_cache_get_oid(const DoutPrefixProvide
           };
 
   bool found_in_cache = true;
-  int ret = -1;
+  int ret = blockDir->get(dpp, &block, y);
+  block.cacheObj.objName = this->get_name(); // account for blocks representing simple requests
```

Comment:
get_oid() will populate the object name correctly when no instance is provided. This will happen as part of the read_permission call which invokes get_obj_attrs(). And later once set_instance is called and the prepare method again invokes this method, the get_oid() method will access the instance specific block entry. so add an entry for a delete marker instance, apart from the one that has the latest version. We can not afford to slow down the read path, as the delete might be called once, but `read` can be called multiple times.